olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.h

```diff
@@ -56,6 +58,21 @@ class PartitionsRepository final {
       client::CancellationContext cancellation_context,
       const DataRequest& data_request, client::OlpClientSettings settings);
 
+  static model::Partition PartitionFromSubQuad(
+      const model::SubQuad& sub_quad, const std::string& partition);
+
+  static PartitionsResponse QueryPartitionsAndGetDataHandle(
+      const client::HRN& catalog, const std::string& layer_id,
+      TileRequest request, int64_t version, client::CancellationContext context,
+      client::OlpClientSettings settings,
+      std::string& requested_tile_data_handle);
+
+  static model::Partitions GetTileFromCache(const client::HRN& catalog,
+                                            const std::string& layer_id,
+                                            TileRequest request,
+                                            int64_t version,
+                                            client::OlpClientSettings settings);
```

Comment:
Why not only `GetVersionedTile()`?
Why do you need three public methods if all three serve actually an internal implementation purpose and are not in any point related to what DataRepository needs?
DataRepository::GetVersionedTile() is only interested in the partition metadata for his request. How you get this partition metadata (by using quad tree index, by requesting single partition) is implementation details of PartitionsRepository and should not be handled by DataRepository.

Reply:
I can move this methods to DataRepository and make them private. It was spitted to do not have very big method, and theoretically this methods could be useful for getting metadata for partitions in future. Also it easier to test and cover all functionality separately. Should I move everything to one function?